### PR TITLE
Update data types (uint8_t and uint16_t)

### PR DESCRIPTION
**Pull request by joaquinbvw**
Some small changes on the main code:

Making the DUCOID calculation just once and storing it on a String object.
-Using uint16_t type instead of unsigned int to be consistent between platforms (some microcontrollers or compilers use 16 bits for unsigned int and others use up to 32)..
- Including while(!Serial); on the Setup section for the Leonardo. I don't have other 8 bit Arduino boards so this should be tested with other Arduinos.
- Including a Serial.flush() on the setup section, this is just in case the computer sends anything to the Arduino before the Arduino starts.
- Moving the declaration of the job array outside the function or the loop so it has a fixed size, I've put the max size someone told me on Discord. If anyone can check it would be nice.
- Changing strlen to length() from the String class so it can give more accurate values, inside the ducos1a function.

Other code files:
- Changing the data type unsigned int to uint16_t and int to uint8_t to be consistent between different platforms and to safe some memory and maybe improve speed. I have checked where I changed to uint8_t that the variables never exceed 255 so it's safe to do this. I haven't checked if the uint16_t can be changed to uint8_t, if this is possible could improve more the code.

I didn't know I could upload more than one file on the same pull request so I was doing it for each file lol. Sorry about that, I will remove the other pull requests and leave this one with all the changes.

### CODE REVIEW

File: Arduino_Code/Arduino_Code.ino [hunk 1]
```diff
@@ -93,18 +98,16 @@ void loop() {
     while (Serial.available())
       Serial.read();
     // Start time measurement
-    unsigned long startTime = micros();
+    uint16_t startTime = micros();
```
Comment (dexter3k) [inline Arduino_Code/Arduino_Code.ino]: micros() result wont fit into uint16_t, an uint32_t is needed here. Otherwise wrong hashrate will be reported.

**Comment by revoxhere** [comment]
As dexter3k suggested, the code reports wrong hashrate at uint16_t, but that's an easy fix. I'm really surprised about the hashrate improvement though - 196 H/s!

**Comment by revoxhere** [comment]
As dexter3k suggested, the code reports wrong hashrate at uint16_t, but that's an easy fix. I'm really surprised about the hashrate improvement though - 196 H/s!

**Comment by joaquinbvw** [comment]
I've fix the uint16_t to uint32_t. Thanks for finding it @dexter3k

**Pull request merged**